src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationService.java

```diff
@@ -1,102 +1,217 @@
+     * @throws ResponseStatusException (204 - NO_CONTENT) if the configuration was not found
+     */
+    public ConfigurationDTO delete(final UUID id) {
+        log.info("delete configuration {}", id);
+        Configuration configuration = configurationRepository
+                .findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NO_CONTENT));
+        configurationRepository.deleteById(id);
+        return configurationMapper.toDTO(configuration);
+    }
+
+    /**
+     * Get all codes of a configuration.
+     *
+     * @param id the configuration id
+     * @return the codes of the specified configuration
+     */
+    public List<CodeDTO> getCodes(final UUID id) {
+        log.debug("get codes from configuration {}", id);
+        final Configuration configuration = getConfiguration(id);
+        return codeMapper.toDTO(configuration.getCodes());
+    }
+
+    public ConfigurationVM getViewModel(UUID id) {
+        log.debug("get configuration view model {}", id);
+        final Configuration configuration = getConfiguration(id);
+        final ConfigurationVM configurationVM = new ConfigurationVM(new ArrayList<>());
+        for (Code code : configuration.getCodes()) {
+            final CodeVM codeVM = new CodeVM(new ArrayList<>());
+            final Solution solution = solutionRepository
+                    .findByCodeId(code.getId())
+                    .orElseThrow(() ->
+                            new ResponseStatusException(
+                                    HttpStatus.NOT_FOUND,
+                                    String.format("Solution with code id %s not found.", code.getId())
+                            )
+                    );
+
+            List<WordVM> row = new ArrayList<>();
+            for (Word word : code.getWords()) {
+                if (word.getWord().equals("\n")) {
+                    codeVM.getWords().add(row);
+                    row = new ArrayList<>();
+                    continue;
+                }
+                Optional<Bug> bug = solution
+                        .getBugs()
+                        .stream()
+                        .filter(b -> b.getWord().getId().equals(word.getId()))
+                        .findFirst();
+                if (word.getWord().equals(" ") && bug.isEmpty()) {
+                    continue;
+                }
+                final WordVM wordVM = new WordVM();
+                if (bug.isPresent()) {
+                    wordVM.setErrorType(bug.get().getErrorType());
+                    wordVM.setCorrectValue(bug.get().getCorrectValue());
+                    wordVM.setDisplayValue(word.getWord());
+                } else {
+                    wordVM.setCorrectValue(word.getWord());
+                }
+                row.add(wordVM);
+            }
+            codeVM.getWords().add(row);
+            configurationVM.getCodes().add(codeVM);
+        }
+
+        return configurationVM;
+    }
```

Comment:
too long method, split it up, add java doc

Reply:
Also, misused `Optional` use.

Reply:
And `equals(" ")` will NOT scale well!
That is error prone AF.
It's probably best to extract it into a method.